apis/python/src/tiledbsoma/_index_util.py

```diff
@@ -3,26 +3,34 @@
 SOMATileDBContext which would otherwise ensue.
 """
 
+from typing import Optional, cast
+
 import numpy as np
 import pandas as pd
+from somacore.query.types import IndexLike
 
 from tiledbsoma import pytiledbsoma as clib
 
 from .options import SOMATileDBContext
 
 
-def build_index(
-    keys: np.typing.NDArray[np.int64], context: SOMATileDBContext
-) -> clib.IntIndexer:
+def tiledbsoma_build_index(
+    keys: np.typing.NDArray[np.int64],
+    *,
+    context: Optional[SOMATileDBContext] = None,
+    thread_count: int = 4,
+) -> IndexLike:
     """Builds an indexer object compatible with :meth:`pd.Index.get_indexer`."""
     if len(np.unique(keys)) != len(keys):
         raise pd.errors.InvalidIndexError(
             "Reindexing only valid with uniquely valued Index objects"
         )
-    tdb_concurrency = int(
-        context.tiledb_ctx.config().get("sm.compute_concurrency_level", 10)
-    )
-    thread_count = tdb_concurrency // 2
+    if context is not None:
+        tdb_concurrency = int(
+            context.tiledb_ctx.config().get("sm.compute_concurrency_level", 10)
+        )
+        thread_count = tdb_concurrency // 2
+
     reindexer = clib.IntIndexer()
     reindexer.map_locations(keys, thread_count)
-    return reindexer
+    return cast(IndexLike, reindexer)
```

Comment:
You can minimize the amount of errors that will be caught with a type-ignore comment:

```
return reindexer  # type: ignore[no-any-return]
```

In this case I think using this particular version of `type: ignore` is better than using a `cast`, since this ends up being more specific about what should be allowed (and it can be dropped if `clib.IntIndexer` gets type annotation data somehow). `cast` is a lot like a blanket `# type: ignore` for an assignment statement.

It is fine to leave it as-is, though; this is an incredibly minor difference.

Reply:
sounds good. I'll use this. I did not know `ignore[no-any-return]` exists ;)